diff --git a/dist/elements/Popup/PopupTrigger/index.js b/dist/elements/Popup/PopupTrigger/index.js
index ae8251799d52a25e97a8d7711ce14e0b46510b16..da9d117ac671dff1d771716c5095e5872f5d6416 100644
--- a/dist/elements/Popup/PopupTrigger/index.js
+++ b/dist/elements/Popup/PopupTrigger/index.js
@@ -1,7 +1,7 @@
 'use client';
 
 import { jsx as _jsx } from "react/jsx-runtime";
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './index.scss';
 const baseClass = 'popup-button';
 export const PopupTrigger = props => {
@@ -15,7 +15,12 @@ export const PopupTrigger = props => {
     setActive,
     size
   } = props;
-  const classes = [baseClass, className, `${baseClass}--${buttonType}`, !noBackground && `${baseClass}--background`, size && `${baseClass}--size-${size}`, disabled && `${baseClass}--disabled`].filter(Boolean).join(' ');
+  const [mounted, setMounted] = useState(false);
+  useEffect(() => {
+    setMounted(true);
+  }, []);
+  const isDisabled = mounted ? Boolean(disabled) : false;
+  const classes = [baseClass, className, `${baseClass}--${buttonType}`, !noBackground && `${baseClass}--background`, size && `${baseClass}--size-${size}`, isDisabled && `${baseClass}--disabled`].filter(Boolean).join(' ');
   const handleClick = () => {
     setActive(!active, false);
   };
@@ -40,7 +45,7 @@ export const PopupTrigger = props => {
   }
   return /*#__PURE__*/_jsx("button", {
     className: classes,
-    disabled: disabled,
+    disabled: isDisabled,
     onClick: handleClick,
     onKeyDown: handleKeyDown,
     tabIndex: 0,
diff --git a/dist/forms/Form/index.js b/dist/forms/Form/index.js
index 0000000000000000000000000000000000000000..1111111111111111111111111111111111111111 100644
--- a/dist/forms/Form/index.js
+++ b/dist/forms/Form/index.js
@@ -717,7 +717,7 @@ export const Form = props => {
           children: /*#__PURE__*/_jsx(SubmittedContext, {
             value: submitted,
             children: /*#__PURE__*/_jsx(InitializingContext, {
-              value: !isMounted || isMounted && initializing,
+              value: isMounted && initializing,
               children: /*#__PURE__*/_jsx(ProcessingContext, {
                 value: processing,
                 children: /*#__PURE__*/_jsx(BackgroundProcessingContext, {
diff --git a/dist/fields/Checkbox/Input.js b/dist/fields/Checkbox/Input.js
index 0000000000000000000000000000000000000000..1111111111111111111111111111111111111111 100644
--- a/dist/fields/Checkbox/Input.js
+++ b/dist/fields/Checkbox/Input.js
@@ -43,7 +43,7 @@ export const CheckboxInput = t0 => {
     t2 = $[1];
   }
   useEffect(t1, t2);
-  const readOnly = readOnlyFromProps || !isHydrated;
+  const readOnly = isHydrated ? Boolean(readOnlyFromProps) : false;
   const t3 = checked && `${inputBaseClass}--checked`;
   const t4 = readOnly && `${inputBaseClass}--read-only`;
   let t5;
@@ -64,7 +64,7 @@ export const CheckboxInput = t0 => {
       "aria-label": "",
       "aria-labelledby": name,
       checked: t7,
-      disabled: readOnly,
+      disabled: readOnly,
       id,
       name,
       onChange: onToggle,
diff --git a/dist/fields/Checkbox/index.js b/dist/fields/Checkbox/index.js
index 0000000000000000000000000000000000000000..1111111111111111111111111111111111111111 100644
--- a/dist/fields/Checkbox/index.js
+++ b/dist/fields/Checkbox/index.js
@@ -2,7 +2,7 @@
 
 import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
 import { rtlLanguages } from '@payloadcms/translations';
-import React, { useCallback, useMemo } from 'react';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import { RenderCustomComponent } from '../../elements/RenderCustomComponent/index.js';
 import { FieldDescription } from '../../fields/FieldDescription/index.js';
 import { FieldError } from '../../fields/FieldError/index.js';
@@ -48,6 +48,10 @@ const CheckboxFieldComponent = props => {
     }
   } = useTranslation();
   const isRTL = rtlLanguages.includes(language);
+  const [isHydrated, setIsHydrated] = useState(false);
+  useEffect(() => {
+    setIsHydrated(true);
+  }, []);
   const memoizedValidate = useCallback((value, options) => {
     if (typeof validate === 'function') {
       return validate(value, {
@@ -106,7 +110,7 @@ const CheckboxFieldComponent = props => {
       name: path,
       onToggle: onToggle,
       partialChecked: partialChecked,
-      readOnly: readOnly || disabled,
+      readOnly: isHydrated ? readOnly || disabled : false,
       required: required
     }), /*#__PURE__*/_jsx(RenderCustomComponent, {
       CustomComponent: Description,
